core/src/main/scala/tech/beshu/ror/accesscontrol/utils/CirceOps.scala

```diff
@@ -304,6 +325,12 @@ object CirceOps {
       downFields(name).asWithError[Option[NonEmptyString]](s"Field $name cannot be empty")
     }
 
+    def downFieldAs[T: Decoder](name: String): Decoder.Result[T] = {
+      value.downField(name).as[T].adaptError {
+        case error: DecodingFailure => error.modifyError(errorMessage => s"Error for field '$name': $errorMessage")
```

Comment:
why do we need to do it?

Reply:
We did not have tests for failure cases. When I added such ones, the error messages were like "Non-empty list of user ID patterns is required". Without the context, it's hard to locate the malformed configuration part. I know it's not an elegant error handling, but it was the simplest way to present some context in the error messages without modifying the domain.